test/e2e/suite/addons/addons_test.go

```diff
@@ -0,0 +1,117 @@
+//go:build e2e
+// +build e2e
+
+package addons
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"math/rand"
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	"github.com/onsi/ginkgo/v2/types"
+	. "github.com/onsi/gomega"
+	"gopkg.in/yaml.v2"
+
+	"github.com/aws/eks-hybrid/test/e2e"
+	"github.com/aws/eks-hybrid/test/e2e/suite"
+)
+
+var (
+	filePath    string
+	suiteConfig *suite.SuiteConfiguration
+)
+
+const numberOfNodes = 1
+
+func init() {
+	flag.StringVar(&filePath, "filepath", "", "Path to configuration")
+}
+
+func TestE2E(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Addon Smoke Test Suite")
+}
+
+var _ = SynchronizedBeforeSuite(
+	func(ctx context.Context) []byte {
+		suiteConfig := suite.BeforeSuiteCredentialSetup(ctx, filePath)
+		test := suite.BeforeVPCTest(ctx, &suiteConfig)
+		credentialProviders := suite.AddClientsToCredentialProviders(suite.CredentialProviders(), test)
+		osList := suite.OSProviderList(credentialProviders)
+
+		// pick 3 random OS/Version/Provider combinations for metricsServer tests worker nodes
+		nodesToCreate := make([]suite.NodeCreate, 0, numberOfNodes)
+
+		rand.Shuffle(len(osList), func(i, j int) {
+			osList[i], osList[j] = osList[j], osList[i]
+		})
+
+		for i := range numberOfNodes {
+			os := osList[i].OS
+			provider := osList[i].Provider
+			nodesToCreate = append(nodesToCreate, suite.NodeCreate{
+				OS:           os,
+				Provider:     provider,
+				InstanceName: test.InstanceName("addon-smoke-test", os, provider),
+				InstanceSize: e2e.Large,
+				NodeName:     fmt.Sprintf("addon-test-node-%s-%s", provider.Name(), os.Name()),
+			})
+		}
+		suite.CreateNodes(ctx, test, nodesToCreate)
+
+		suiteJson, err := yaml.Marshal(suiteConfig)
+		Expect(err).NotTo(HaveOccurred(), "suite config should be marshalled successfully")
+		return suiteJson
+	},
+	// This function runs on all processes, and it receives the data from
+	// the first process (a json serialized struct)
+	// The only thing that we want to do here is unmarshal the data into
+	// a struct that we can make accessible from the tests. We leave the rest
+	// for the per tests setup code.
+	func(ctx context.Context, data []byte) {
+		suiteConfig = suite.BeforeSuiteCredentialUnmarshal(ctx, data)
+	},
+)
+
+var _ = Describe("Hybrid Nodes", func() {
+	When("using peered VPC", func() {
+		var addonEc2Test *suite.AddonEc2Test
+		var createSucceeded bool
```

Comment:
you can define this in the It scope, it doesn't need to be up here. this will be shared by other It blocks, and i don't think that's necessary